internal/controller/cloud-resources/aws/awsnfsbackupschedule_test.go

```
@@ -4,67 +4,94 @@ import (
-	var nfsInstance *cloudcontrolv1beta1.NfsInstance
-	var awsNfsId string
-	var scope *cloudcontrolv1beta1.Scope
 
-	awsAccountId := "974658265573"
+	It("Scenario: Creates recurring backup schedule with existing backups", func() {
+		var (
+			suffix             string
+			skrNfsVolumeName   string
+			nfsInstanceName    string
+			scheduleName       string
+			existingBackupName string
+			skrNfsVolume       *cloudresourcesv1beta1.AwsNfsVolume
+			nfsInstance        *cloudcontrolv1beta1.NfsInstance
+			scope              *cloudcontrolv1beta1.Scope
+			backupSchedule     *cloudresourcesv1beta1.AwsNfsBackupSchedule
+			existingBackup     *cloudresourcesv1beta1.AwsNfsVolumeBackup
+		)
 
-	BeforeEach(func() {
-		By("Given KCP Scope exists", func() {
-			now = time.Now().UTC()
-			skrNfsVolumeName = uuid.NewString()
-			skrNfsVolume = &cloudresourcesv1beta1.AwsNfsVolume{}
-			nfsInstanceName = uuid.NewString()
-			nfsInstance = &cloudcontrolv1beta1.NfsInstance{}
-			awsNfsId = uuid.NewString()
-			scope = &cloudcontrolv1beta1.Scope{}
+		// This ensures each parallel test run has different resource names
+		suffix = fmt.Sprintf("recurring-%d", GinkgoParallelProcess())
+		skrNfsVolumeName = fmt.Sprintf("aws-vol-%s", suffix)
+		nfsInstanceName = fmt.Sprintf("aws-inst-%s", suffix)
+		scheduleName = fmt.Sprintf("aws-schedule-%s", suffix)
+		existingBackupName = fmt.Sprintf("aws-backup-existing-%s", suffix)
+
+		// Initialize objects
+		skrNfsVolume = &cloudresourcesv1beta1.AwsNfsVolume{}
+		nfsInstance = &cloudcontrolv1beta1.NfsInstance{}
+		scope = &cloudcontrolv1beta1.Scope{}
+		backupSchedule = &cloudresourcesv1beta1.AwsNfsBackupSchedule{}
+		existingBackup = &cloudresourcesv1beta1.AwsNfsVolumeBackup{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      existingBackupName,
+				Namespace: DefaultSkrNamespace,
+			},
+		}
+
+		DeferCleanup(func() {
+			By("Cleanup: Delete test resources", func() {
+				_ = Delete(infra.Ctx(), infra.SKR().Client(), backupSchedule)
```

We do no want to ignore errors
```suggestion
				Expect(Delete(infra.Ctx(), infra.SKR().Client(), backupSchedule)).To(Succeed())
```
Ideally you would also ensure it was really deleted and not leaking out of scenario with
```suggestion
				Eventually(IsDeleted).WithArguments(infra.Ctx(), infra.SKR().Client(), backupSchedule).Should(Succeed())
```